codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/UnstableDerive.kt

```diff
@@ -0,0 +1,17 @@
+object UnstableDerive {
+    // double `#` is because this data is passed onto writeInLine, which will interpret it as a variable with single `#`
+    val AttrUnstableSerdeAny = """##[cfg(all(aws_sdk_unstable, any(feature = "serde-serialize", feature = "serde-deserialize")))]"""
+    val AttrUnstableSerdeBoth = """
+        all(aws_sdk_unstable, all(feature = "serde-serialize", feature = "serde-deserialize"))
+    """
+    val AttrUnstableSerialize = """
+        all(aws_sdk_unstable, feature = "serde-serialize")
+    """
+    val AttrUnstableDeserialize = """
+        all(aws_sdk_unstable, feature = "serde-deserialize")
+    """
+
+    // double `#` is because this data is passed onto writeInLine, which will interpret it as a variable with single `#`
+    val UnstableDerive = """##[cfg_attr(all(aws_sdk_unstable, feature = "serde-serialize"), derive(serde::Serialize))]
+##[cfg_attr(all(aws_sdk_unstable, feature = "serde-deserialize"), derive(serde::Deserialize))]"""
+}
```

Comment:
We have a new way of constructing attributes and you can add these to the `Attribute` class' companion object. Here's what that would look like:

```kotlin
        fun cfg_attr(vararg attrMacros: Writable): Writable = macroWithArgs("cfg_attr", *attrMacros)
        val UnstableSerdeAny = Attribute(cfg(all(writable("aws_sdk_unstable"), any(pair("feature" to "serde-serialize"), pair("feature" to "serde-deserialize")))))
        val UnstableSerdeAll = all(writable("aws_sdk_unstable"), all(pair("feature" to "serde-serialize"), pair("feature" to "serde-deserialize")))
        val UnstableSerialize = all(writable("aws_sdk_unstable"), pair("feature" to "serde-serialize"))
        val UnstableDeserialize = all(writable("aws_sdk_unstable"), pair("feature" to "serde-deserialize"))
        val UnstableSerdeDerive = writable {
            Attribute(cfg_attr(all(writable("aws_sdk_unstable"), pair("feature" to "serde-serialize"), derive(CargoDependency.Serde.resolve("Serialize"))))).render(this)
            Attribute(cfg_attr(all(writable("aws_sdk_unstable"), pair("feature" to "serde-deserialize"), derive(CargoDependency.Serde.resolve("Deserialize"))))).render(this)
        }
```

Because we actually reference the `CargoDependency` for `serde` in `UnstableDerive`, we'll automatically get it added to the `Cargo.toml` whenever we write `UnstableDerive`.

All of these are `Writable` and you can write them by calling `.render(this)` where `this` is a rust writer:

```kotlin
import software.amazon.smithy.rust.codegen.core.rustlang.Attribute

// ...
private fun someRenderFun(writer: RustWriter) {
    Attribute.UnstableSerdeDerive.render(writer)
}
```

Reply:
Thanks for the help!